### PR TITLE
Support 'ValueTasks' in the 'make method async' fixer.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/17470

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide

File: src/Features/Core/Portable/MakeMethodAsynchronous/AbstractMakeMethodAsynchronousCodeFixProvider.cs [hunk 1]
```diff
@@ -130,8 +130,9 @@ private SyntaxNode GetContainingFunction(Diagnostic diagnostic, CancellationToke
             var compilation = await document.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
             var taskType = compilation.GetTypeByMetadataName("System.Threading.Tasks.Task");
             var taskOfTType = compilation.GetTypeByMetadataName("System.Threading.Tasks.Task`1");
+            var valueTaskOfTType = compilation.GetTypeByMetadataName("System.Threading.Tasks.ValueTask`1");
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/MakeMethodAsynchronous/AbstractMakeMethodAsynchronousCodeFixProvider.cs]: Does the compiler have a "is task like" API we should be using, or are we just doing this for simplicity?

Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/MakeMethodAsynchronous/AbstractMakeMethodAsynchronousCodeFixProvider.cs]: not currently.  Seems like something worthwhile!

Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/MakeMethodAsynchronous/AbstractMakeMethodAsynchronousCodeFixProvider.cs]: I will file bug on this.

Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/MakeMethodAsynchronous/AbstractMakeMethodAsynchronousCodeFixProvider.cs]: Tracking bug here: https://github.com/dotnet/roslyn/issues/17584

**Pull request merged**